ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1299,10 +1299,10 @@ void checkAndMaybeComplete() {
         }
       }
       if (done) {
-        // the chunk is complete now. We can charge against quota for the chunk.
-        if (quotaChargeCallback != null) {
+        // the chunk is complete now. We can charge against quota for the chunk if its not a metadata chunk.
+        if (quotaChargeCallback != null && !(this instanceof MetadataPutChunk)) {
```

Comment:
we should also check if the chunkException is null or not, just like what we did in the GetBlobOperation.